docs/control-structures.rst

```diff
@@ -70,15 +70,15 @@ only (locally) sets the value and amount of gas sent with the function call and
 parentheses at the end perform the actual call.
 
 .. warning::
-    Any interaction with another contract imposes a certain danger, especially
+    Any interaction with another contract imposes a potential danger, especially
     if the source code of the contract is not known in advance. The current
-    contract hands over control to the called contract and that might do
-    just about anything. Be prepared that it calls into other contracts of
-    your system and perhaps even back into the calling contract before your
+    contract hands over control to the called contract and that may potentially
+    do just about anything. Be prepared in case it calls into other contracts of
```

Comment:
We should probably mention polymorphism here and contracts only adhering to an interface but that their implementation can do arbitrary other things.